channeld/channeld.c

```diff
@@ -2270,7 +2295,11 @@ static struct commitsig_info *handle_peer_commit_sig(struct peer *peer,
 		tal_steal(commitsigs, result);
 	}
 
-	assert(old_secret);
+	// If the HSM doesn't support WIRE_HSMD_REVOKE_COMMITMENT_TX we'd better
+	// have the old_secret at this point.
+	if (HSM_MAX_VERSION < 5 ||
+	    !hsm_is_capable(peer->hsm_capabilities, WIRE_HSMD_REVOKE_COMMITMENT_TX))
```

Comment:
Mh this is just a comment for a future improvement. Would be nice to have a method to avoid checking the version here, but with the current enum system for the `hsm_capabilities` this seems hard